### PR TITLE
Fixed flaky tests

### LINKED ISSUE #455: intermittent test failures in community build (Format, StableVersion)
example of `Format.formatLowercase` failing (sometimes it's `Format.formatUppercase` that fails instead):

* Format.formatLowercase

example of StableVersion failing: 

* https://scala-ci.typesafe.com/job/scala-2.13.x-integrate-community-build/3046/artifact/logs/giter8-build.log

they fail only intermittently, but pretty commonly, or I wouldn't be here filing the bug

I will exclude the tests from the community build. (I can re-enable them on request, if a fix is believed to have been found.)

### PR DESCRIPTION
**Pull request by TonioGela**
Closes #455.

See [this comment](https://github.com/foundweekends/giter8/issues/455#issuecomment-855447490) for details.

### CODE REVIEW

**Comment by TonioGela** [comment]
ofc the "Fixed flaky tests" PR has flaky tests

**Comment by TonioGela** [comment]
I would really like to spot the problem, but testing locally 100 times with jdk 11 I haven't been able to reproduce it. Any idea @eed3si9n?

**Comment by eed3si9n** [comment]
Maybe the property based testing doesn't need to consider version numbers with hanzi/kanji characters in them?

```
[info] ! StableVersion.ordering: Falsified after 943 passed tests.
[info] > Labels of failing property: 
[info] Reversing and sorting should be the same as sorting. Expected version: 365.181.24.0-臿揧氛-䗶蒳쬍㣴ꂦ-弮堢䋊弑⯄೏즃㴼㼺䀧᪟⿽, but got: 365.181.24-㦧녦-㓒ꑡ-Ყ懸喺݄辺芭ᤀ
```

**Comment by TonioGela** [comment]
Wow, how did you picked that line in that huge log? Btw, I think you're right, in a few I'll push a fix for that

**Comment by TonioGela** [comment]
Is not so trivial, after some iterations I get errors like this
```
failing seed for StableVersion.ordering is 7cjg5dZl9xbfN9xQFEJ2dWFH63NCIWZuWFMC77gKS2C=
[info] ! StableVersion.ordering: Falsified after 20155 passed tests.
[info] > Labels of failing property:
[info] Reversing and sorting should be the same as sorting. Expected version: 319.975.54.0-J4-im1-iquf4jladvx, but got: 319.975.54-V50uE2WjP1
[info] > ARG_0: List(319.975.54.0-J4-im1-iquf4jladvx, 319.975.54-V50uE2WjP1)
```
even if I set something strict like
```scala
private val genChar = Gen.alphaNumChar
```
I'll investigate further

**Comment by TonioGela** [comment]
I don't know why but in [this](https://github.com/foundweekends/giter8/commit/4f9354af023793dc341377ba1fda8e5e3618327c) commit [this line](https://github.com/foundweekends/giter8/blob/develop/library/src/test/scala/giter8/VersionNumberOrderingTest.scala#L88) was added:
```scala
numbers = prefix.toSeq ++: vn.numbers.drop(prefix.length)
```
To the best of my understanding, this should concatenate and empty list to `prefix.toSeq`, but sometimes the net effect is that lists like `List(83, 39, 374)` became `List(83, 39, 374, 0)` and so the test fails.

Removing this `map` the test passed flawlessly
```scala
genVersionNumber.map { vn =>
          VersionNumber(
            numbers = prefix.toSeq ++: vn.numbers.drop(prefix.length),
            tags = vn.tags,
            extras = vn.extras
          )
        }
```

**Comment by eed3si9n** [comment]
In general feel free to remove any flaky tests too if it's not delivering values. It's not like we have a ton of bug reports complaining about version number parsing.

**Comment by TonioGela** [comment]
> In general, feel free to remove any flaky tests too if it's not delivering values. It's not like we have a ton of bug reports complaining about version number parsing.

Okay 👍 This time, I think I fixed it in a meaningful way btw :)

**Review by eed3si9n — APPROVED** [review]

File: justfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+clean:
+    -rm -rf /Users/agelameris/.g8/boot/org.foundweekends.giter8/giter8_2.12/0.13.1-SNAPSHOT
+    -rm -rf /Users/agelameris/.ivy2/local/org.foundweekends.giter8/sbt-giter8/scala_2.12/sbt_1.0/0.13.1-SNAPSHOT/
```
Comment (eed3si9n) [inline justfile]: Was this file meant to be included into the PR?

Comment (TonioGela) [reply justfile]: No, it's my bad, sorry. I'll delete it.

Comment (TonioGela) [reply justfile]: I had to push force ofc, sorry for history rewriting

**Pull request merged**